src/Insight.test.ts

```diff
@@ -0,0 +1,41 @@
+/* tslint:disable:no-relative-imports */
+import { describe } from 'riteway'
+
+import { throwError, InsightError, BlockHeightOutOfRangeError } from './Insight'
+
+describe('throwError', async (should: any) => {
+  const { assert } = should()
+
+  {
+    const error = () => {
+      try {
+        return throwError('random')
+      } catch (e) {
+        return e
+      }
+    }
+
+    assert({
+      given: 'random string',
+      should: 'throw new InsightError with random string',
+      actual: error(),
```

Comment:
Instead of creating the function above that sort of pulls the logic away from the assertion, you can use riteways `Try`

```
assert({
  given: 'random string',
  should: 'throw new InsightError with random string',
  actual: Try(throwError, 'ramdom string'),
  expected: new InsightError('random'),
})
```

Reply:
See the example in the riteway readme for details on importing and using `Try`, the last assertion uses it
https://github.com/ericelliott/riteway

Reply:
Just approved and then saw this comment. Didn't know about Riteway's Try — definitely looks like a better approach.